python/SAGEMAKER_NOTEBOOK_KMS_CONFIGURED/SAGEMAKER_NOTEBOOK_KMS_CONFIGURED_test.py

```diff
@@ -0,0 +1,275 @@
+    described_notebook_instances = [{'NotebookInstanceName': 'trial12', 'KmsKeyId': 'arn:aws:kms:us-east-1:075975677358:key/7af97db7-f6a3-4d0a-87b9-a2737b54856d'}, {'NotebookInstanceName': 'trial123', 'KmsKeyId': 'arn:aws:kms:us-east-1:075975677358:key/7af97db7-f6a3-4d0a-87b9-a2737b54856d'}]
+    main_rule_params = '{"keyIds":"arn:aws:kms:us-east-1:075975677358:key/7af97db6-f6a3-4d0a-87b9-a2737b54856d"}'
+    matched_key = '{"keyIds":"arn:aws:kms:us-east-1:075975677358:key/7af97db7-f6a3-4d0a-87b9-a2737b54856d"}'
+
+    notebook_instances_list_mixed = [{'NotebookInstances': [{'NotebookInstanceName': 'trial12', 'NotebookInstanceArn': 'arn:aws:sagemaker:us-east-1:075975677358:notebook-instance/trial12'}, {'NotebookInstanceName': 'trial123', 'NotebookInstanceArn': 'arn:aws:sagemaker:us-east-1:075975677358:notebook-instance/trial123'}, {'NotebookInstanceName': 'trial1234', 'NotebookInstanceArn': 'arn:aws:sagemaker:us-east-1:075975677358:notebook-instance/trial1234'}]}]
+    described_notebooks_mixed = [{'NotebookInstanceName': 'trial12', 'KmsKeyId': 'arn:aws:kms:us-east-1:075975677358:key/7af97db7-f6a3-4d0a-87b9-a2737b54856d'}, {'NotebookInstanceName': 'trial123', 'KmsKeyId': 'arn:aws:kms:us-east-1:075975677358:key/7af97db6-f6a3-4d0a-87b9-a2737b54856d'}, {'NotebookInstanceName': 'trial1234'}]
+
+    def test_scenario_1_invalid_param(self):
+        rule_param = '{"keyIds":"83de41d66530-49c1-9cb7-1de1560ce5tg"}'
+
+        #SAGEMAKER_CLIENT_MOCK.list_notebook_instances = MagicMock(return_value=self.instance_list)
+        SAGEMAKER_CLIENT_MOCK.configure_mock(**{
+            "get_paginator.return_value": SAGEMAKER_CLIENT_MOCK,
+            "paginate.return_value": [self.notebook_instances_list]})
+        SAGEMAKER_CLIENT_MOCK.describe_notebook_instance = MagicMock(side_effect=self.described_notebook_instances)
+        print(rule_param)
+        lambda_event = build_lambda_scheduled_event(rule_parameters=rule_param)
+        print(lambda_event)
+        response = RULE.lambda_handler(lambda_event, {})
+        print(response)
+        assert_customer_error_response(self, response, 'InvalidParameterValueException', 'The KMS Key id should be in the right format.')
+
+
+    def test_scenario_2_no_instance(self):
+        rule_param = '{"KeyIds":"arn:aws:kms:us-east-1:075975677358:key/7af97db7-f6a3-4d0a-87b9-a2737b54856d"}'
+        notebook_instances_list2 = {"NotebookInstances": []}
+        described_notebook_instances2 = {}
+
+        #SAGEMAKER_CLIENT_MOCK.list_notebook_instances = MagicMock(return_value=self.instance_list)
+        SAGEMAKER_CLIENT_MOCK.configure_mock(**{
+            "get_paginator.return_value": SAGEMAKER_CLIENT_MOCK,
+            "paginate.return_value": [notebook_instances_list2]})
+        SAGEMAKER_CLIENT_MOCK.describe_notebook_instance = MagicMock(side_effect=described_notebook_instances2)
```

Comment:
No need to mock describe call here as it isn't used when there are no notebook instances.